libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/car/extensions/GlideExtensions.kt

```diff
@@ -0,0 +1,30 @@
+package com.mapbox.androidauto.internal.car.extensions
+
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import com.bumptech.glide.RequestManager
+import com.bumptech.glide.request.target.CustomTarget
+import com.bumptech.glide.request.transition.Transition
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+
+internal suspend inline fun RequestManager.request(uri: Uri): Drawable? {
```

Comment:
IMO you should move all this to a class. And then write tests for it. For example, I'm not sure if it should really call `invalidate` after every image. What if the list should be invalidated in "batches" 🤔 .

``` kotlin
class CarFeedbackIconDownloader(val screen: Screen) {
    private val downloadedImages = hashMapOf<Uri, CarIcon>()

    fun download(icon: CarFeedbackIcon): CarIcon {
        return when (icon) {
            is CarFeedbackIcon.Local -> icon.icon
            is CarFeedbackIcon.Remote -> downloadedImages.getOrPut(icon.uri) {
                screen.lifecycleScope.launch { downloadImage(icon.uri) }
                CarIcon.ERROR
            }
        }
    }

    private suspend fun downloadImage(uri: Uri) {
        val resource = Glide.with(screen.carContext).request(uri) ?: return
        val icon = IconCompat.createWithBitmap(resource.toBitmap())
        downloadedImages[uri] = CarIcon.Builder(icon).build()
        screen.invalidate()
    }

    internal suspend inline fun RequestManager.request(uri: Uri): Drawable? {
        return suspendCancellableCoroutine { continuation ->
            val target = object : CustomTarget<Drawable>() {

                override fun onLoadFailed(errorDrawable: Drawable?) {
                    continuation.resume(value = null)
                }

                override fun onResourceReady(resource: Drawable, transition: Transition<in Drawable>?) {
                    continuation.resume(resource)
                }

                override fun onLoadCleared(placeholder: Drawable?) {
                    // Intentionally empty
                }
            }
            continuation.invokeOnCancellation { clear(target) }
            load(uri).into(target)
        }
    }
}
```

Reply:
Also this was just a quick suggestion ^. Honestly not a fan the screen being a parameter - but it also makes the usage nice so it's good too.